core/src/main/java/org/apache/gravitino/cache/BaseEntityCache.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.gravitino.meta.RoleEntity;
+import org.apache.gravitino.meta.SchemaEntity;
+import org.apache.gravitino.meta.TableEntity;
+import org.apache.gravitino.meta.TagEntity;
+import org.apache.gravitino.meta.TopicEntity;
+import org.apache.gravitino.meta.UserEntity;
+import org.apache.gravitino.storage.relational.RelationalEntityStore;
+
+/**
+ * An abstract class that provides a basic implementation for the MetaCache interface. This class is
+ * abstract and cannot be instantiated directly, it is designed to be a base class for other meta
+ * cache implementations.
+ *
+ * <p>The purpose of the BaseMetaCache is to provide a unified way of accessing entity stores,
+ * allowing subclasses to focus on caching logic without having to deal with entity store
+ * management.
+ */
+public abstract class BaseEntityCache implements EntityCache {
+  private static final Map<Entity.EntityType, Class<?>> ENTITY_CLASS_MAP;
+  // The entity store used by the cache, initialized through the constructor.
+  protected final RelationalEntityStore entityStore;
+  protected final CacheConfig cacheConfig;
+
+  static {
+    Map<Entity.EntityType, Class<?>> map = new EnumMap<>(Entity.EntityType.class);
+    map.put(Entity.EntityType.METALAKE, BaseMetalake.class);
+    map.put(Entity.EntityType.CATALOG, CatalogEntity.class);
+    map.put(Entity.EntityType.SCHEMA, SchemaEntity.class);
+    map.put(Entity.EntityType.TABLE, TableEntity.class);
+    map.put(Entity.EntityType.FILESET, FilesetEntity.class);
+    map.put(Entity.EntityType.MODEL, ModelEntity.class);
+    map.put(Entity.EntityType.TOPIC, TopicEntity.class);
+    map.put(Entity.EntityType.TAG, TagEntity.class);
+    map.put(Entity.EntityType.MODEL_VERSION, ModelVersionEntity.class);
+    map.put(Entity.EntityType.COLUMN, ColumnEntity.class);
+    map.put(Entity.EntityType.USER, UserEntity.class);
+    map.put(Entity.EntityType.GROUP, Entity.class);
+    map.put(Entity.EntityType.ROLE, RoleEntity.class);
+    ENTITY_CLASS_MAP = Collections.unmodifiableMap(map);
+  }
+
+  /**
+   * Returns the class of the entity based on its type.
+   *
+   * @param type The entity type
+   * @return The class of the entity
+   * @throws IllegalArgumentException if the entity type is not supported
+   */
+  @SuppressWarnings("unchecked")
+  public static <E extends Entity & HasIdentifier> Class<E> getEntityClass(Entity.EntityType type) {
+    Preconditions.checkNotNull(type, "EntityType must not be null");
+
+    Class<?> clazz = ENTITY_CLASS_MAP.get(type);
+    if (clazz == null) {
+      throw new IllegalArgumentException("Unsupported EntityType: " + type.getShortName());
+    }
+
+    return (Class<E>) clazz;
+  }
+
+  /**
+   * Returns the {@link NameIdentifier} of the metadata based on its type.
+   *
+   * @param metadata The entity
+   * @return The {@link NameIdentifier} of the metadata
+   */
+  protected static NameIdentifier getIdentFromMetadata(Entity metadata) {
```

Comment:
Change the variable name to `entity`, `entity` is an entity, `metadata` is just a concept.